awx/main/tests/unit/test_tasks.py

```diff
@@ -1420,6 +1429,46 @@ def run_pexpect_side_effect(*args, **kwargs):
         self.run_pexpect.side_effect = run_pexpect_side_effect
         self.task.run(self.pk)
 
+    @pytest.mark.parametrize('with_credential', [True, False])
+    def test_custom_source(self, with_credential):
```

Comment:
This tests that custom inventory scripts can make use of environment variables from cloud credentials.